### ISSUE TITLE
HP 15-bs183no

### ISSUE THREAD

**Issue by viftegalskap** [hirschmann/nbfc#652]
I've been trying to create a config file for this laptop, but I can't seem to get it quite right. Found out (by studying the EC register map and running stress test) that the fan speed reading is in offset 17 (decimal). However, I could see no register that appeared to be the write register for the fan speed.

Existing configs: I looked through all the existing configs and found two other HP laptops (HP Compaq 15-s103tx.xml and HP ENVY m6 1206dx.xml) that also have read register 17. They both have write register 20. I tried both of these configs, but they don't work quite as I had hoped for. The biggest issue seems to be that the fan won't start at all if nbfc is enabled before the fan has started running. If I enable nbfc when the fan is already running, then it seems to work. Then I can see how the values are written to register 20 and reproduced in register 17. However it only works in "auto" mode. When I try to slide the fan % manually in nbfc, nothing happens.

Unable to write manually to EC: After finding out from the existing configs that the write register must be at offset 20, I tried writing values to it in RWeverytning but nothing happened. The EC ignores and immediately overwrites my inputs, and thus I am not able to find the minimum fan speed value. (The maximum I have found from looking at register 17 while stressing the cpu).

I read in the wiki section that: "On some ECs, you must find a register which allows you to set the fan control mode from automatic (= controlled by the EC's firmware) to manual, otherwise the EC will ignore or overwrite external input." So I guess that register (and the appropriate values) must somehow be found to make this work properly?

Any tip to help me get further will be appreciated.

I/O chip: ENE KB9022Q D (I have not been able to find the datasheet).

**Comment by mrDevGo** [comment]
Me too need konfig

**Comment by webloft** [comment]
Try to extract the DSDT ACPI table. I found the register offsets via this table. Also all HP 15-xxxxx seem to be controlled via FRPM, FNTV and RSTV (names in the table).

**Comment by Overc1ocker** [comment]
Been interested in playing around with a DA0014DX hp laptop. I just dumped the SSDTs from the laptop. The second ssdt shows FRPM, FNTV , and RSTV! Any idea how to control the fan? I have attached the SSDT.dsl here.
[ssdt-da0xxx-KB9022QD-fan?.txt](https://github.com/hirschmann/nbfc/files/6059323/ssdt-da0xxx-KB9022QD-fan.txt)

Would love to know where the registers for the temp sensors are as well! Thanks!
